src/Compilers/CSharp/Test/Semantic/Semantics/NullCoalescingAssignmentTests.cs

```diff
@@ -0,0 +1,1408 @@
+﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+using System.Linq;
+using Microsoft.CodeAnalysis.CSharp.Syntax;
+using Microsoft.CodeAnalysis.CSharp.UnitTests;
+using Microsoft.CodeAnalysis.Test.Utilities;
+using Xunit;
+
+namespace Microsoft.CodeAnalysis.CSharp.Semantic.UnitTests.Semantics
+{
+    [CompilerTrait(CompilerFeature.NullCoalescingAssignment)]
+    public class NullCoalescingAssignmentTests : CompilingTestBase
+    {
+        [Fact]
+        public void LocalLvalue()
+        {
+
+            CompileAndVerify(@"
+using System;
+public class C
+{
+    public static void Main()
+    {
+        int? i1 = null;
+        Console.WriteLine(i1 ??= GetInt());
+    }
+    static int GetInt()
+    {
+        Console.WriteLine(""In GetInt"");
+        return 0;
+    }
+}
+", expectedOutput: @"
+In GetInt
+0
+").VerifyIL("C.Main()", @"
+{
+  // Code size       43 (0x2b)
+  .maxstack  1
+  .locals init (int? V_0)
+  IL_0000:  ldloca.s   V_0
+  IL_0002:  initobj    ""int?""
+  IL_0008:  ldloc.0
+  IL_0009:  stloc.0
```

Comment:
IL 8 and 9 seems to do nothing. Could be optimized later.

Reply:
Adding a prototype comment to come back to this.

---
In reply to: [200788151](https://github.com/dotnet/roslyn/pull/28253#discussion_r200788151) [](ancestors = 200788151)